app/controllers/api/v1/entity_controller.rb

```diff
@@ -0,0 +1,194 @@
+class Api::V1::EntityController < Api::V1::BaseApiController
+  include JsonCollectionPagination
+  include DownloadableCsv
+  include AllowFiltering
+
+  before_action :set_model, only: [:show]
+
+  def self.model
+    fail Exception.new("Abstract method")
+  end
+
+  def index
+    index_query
+    index_includes
+    respond_to do |format|
+      format.json { render paginated_json_collection(@collection).merge({ scope: render_scope, each_serializer: render_serializer }) }
+      format.geojson { render paginated_geojson_collection(@collection).merge({ scope: render_scope }) }
+      format.csv { return_downloadable_csv(@collection, self.class.model.name.underscore.pluralize) }
+    end
+  end
+
+  def show
+    respond_to do |format|
+      format.json { render json: @model, serializer: render_serializer, scope: render_scope }
+      format.geojson { render json: @model, serializer: GeoJSONSerializer }
+    end
+  end
+
+  private
+
+  def index_query
+    # Entity
+    @collection = (self.class.model).where('')
+    @collection = AllowFiltering.by_onestop_id(@collection, params)
+    @collection = AllowFiltering.by_tag_keys_and_values(@collection, params)
+    @collection = AllowFiltering.by_updated_since(@collection, params)
+
+    # Geometry
+    index_query_geometry
+
+    # Imported From Feed
+    if params[:imported_from_feed].present?
+      @collection = @collection.where_imported_from_feed(Feed.find_by_onestop_id(params[:imported_from_feed]))
```

Comment:
That might explain why when I run a local request like http://localhost:3000/api/v1/stops?imported_from_feed=f-a-no (with a non-existent feed Onestop ID), I get back:

```json
{"message":"[NoMethodError] undefined method `id' for nil:NilClass"}
```